app/forms/current_school_form.rb

```diff
@@ -0,0 +1,58 @@
+class CurrentSchoolForm < Form
+  attribute :school_search
+  attribute :current_school_id
+
+  attr_reader :schools
+
+  validates :current_school_id, presence: {message: ->(object, data) { object.i18n_errors_path("select_the_school_you_teach_at") }}
+  validate :current_school_must_be_open, if: -> { current_school_id.present? }
+
+  def initialize(claim:, journey:, params:)
+    super
+
+    load_schools
+    self.current_school_id = permitted_params[:current_school_id]
+  end
+
+  def backlink_path
+    params[:slug] if params[:school_search]
+  end
+
+  def save
+    return false unless valid?
+
+    claim.update!({eligibility_attributes: {current_school_id: current_school_id}})
+  end
+
+  def current_school_name
+    claim.eligibility.current_school_name
+  end
+
+  def i18n_errors_path(msg)
```

Comment:
This can probably go in the `Form` base class, and can probably be a private method.